ServerMilestone1.0

NikolaRusakov opened the pull request.

NikolaRusakov committed:
server - Resolved issue, /get-all returns all containers,
OsCheck class uses apache commons sys check now
client - card updated, views renamed to pages,
script section in package.json moved on top

NikolaRusakov committed:
server - /environment works
client - card updated

TODO:
server - in EnvironmentService.java fix computeifpresent
client - col,rows

NikolaRusakov committed:
server - /environment works
client - card updated

TODO:
server - in EnvironmentService.java fix computeifpresent
client - col,rows

petrleocompel commented:
resolves #1

petrleocompel commented on server/src/main/java/cz/plc/prx/docker/dash/service/EnvironmentService.java:
```diff
@@ -12,7 +17,27 @@
-        return null;
+        List<Container> containerList = dcService.getDefaultConnection().listContainersCmd().withShowAll(true).exec();
+        List<Container> sifted = new ArrayList<Container>();
+        for (int i = 0; i < containerList.size(); i++) {
```
Replace with foreach see [example](https://docs.oracle.com/javase/tutorial/displayCode.html?code=https://docs.oracle.com/javase/tutorial/java/nutsandbolts/examples/EnhancedForDemo.java)

petrleocompel commented on server/src/main/java/cz/plc/prx/docker/dash/service/EnvironmentService.java:
```diff
@@ -12,7 +17,27 @@
+            final int in = i;
+
+            InspectContainerResponse exec = dcService.getDefaultConnection().inspectContainerCmd(containerList.get(i).getId()).exec();
+
```
Missing check for blacklisting
```java
if (labels.contains(LABEL_BLACKLIST) && Boolean.valueOf(labels.get(LABEL_BLACKLIST))) {
    continue;
}
```

petrleocompel commented on server/src/main/java/cz/plc/prx/docker/dash/service/EnvironmentService.java:
```diff
@@ -12,7 +17,27 @@
+
+            InspectContainerResponse exec = dcService.getDefaultConnection().inspectContainerCmd(containerList.get(i).getId()).exec();
+
+            exec.getConfig().getLabels().computeIfPresent("com.docker.compose.project", (s, s2) -> {
```
"com.docker.compose.project" should be also as constant

petrleocompel commented on server/src/main/java/cz/plc/prx/docker/dash/service/EnvironmentService.java:
```diff
@@ -12,7 +17,27 @@
+        List<Instance> obj = objectFactory.convert(sifted, Instance.class);
+        List<Environment> environments = new ArrayList<Environment>();
+        obj.forEach(instance -> {
+            environments.add(new Environment().addServicesItem(instance));
```
This is bad idea. We want to know what conatiners are together and we have to make a map of containers with compose project as key. 

So for-loop of container you should make it some-like this
```java

//before loop
Map<String, List<Container>> composeProjects; // may HashMap
//in loop
if (labels.contains(LABEL_COMPOSE_PROJECT)) {
    String composeProjectKey = labels.get(LABEL_COMPOSE_PROJECT);
    if (!composeProjects.contains(composeProjectKey)) {
         composeProjects.put(composeProjectKey, new ArrayList<Container>());
    }
    composeProjects.get(composeProjectKey).add(container);
}
```

petrleocompel commented on server/src/main/java/cz/plc/prx/docker/dash/service/InstanceService.java:
```diff
@@ -24,8 +24,12 @@
     public List<Instance> getAll() {
-        List<Container> containerList = dcService.getDefaultConnection().listContainersCmd().exec();
-        System.out.println(containerList);
+        List<Container> containerList = dcService.getDefaultConnection().listContainersCmd().withShowAll(true).exec();
```
You have to return only container list. 
Now mixing with running containers duplicating some of them.

petrleocompel commented on server/src/main/java/cz/plc/prx/docker/dash/util/OsCheck.java:
```diff
@@ -1,44 +1,17 @@
 
-    // cached result of OS detection
-    protected static OSType detectedOS;
+public class OsCheck {
```
Delete whole class and use Apache Util inplace which is needed.

petrleocompel commented on server/src/main/java/cz/plc/prx/docker/dash/service/DockerConnectionService.java:
```diff
@@ -29,23 +33,27 @@ public DockerClient getDefaultConnection() {
             DockerClientConfig config = null;
 
-            OsCheck.OSType ostype = OsCheck.getOperatingSystemType();
+            String ostype = OsCheck.getOperatingSystemType();
```
Use Apache util for detection

petrleocompel commented on server/src/main/java/cz/plc/prx/docker/dash/service/DockerConnectionService.java:
```diff
@@ -29,23 +33,27 @@ public DockerClient getDefaultConnection() {
-                case MacOS:
-                    throw new NotSupportedException("OS Not Supported YET");
-                case Linux:
+                case "MacOS":
```
Change to `if` and use Apache util for detection

petrleocompel reviewed (changes requested):
Fix issue with duplicating containers in `InstanceService` 
Fix `EnvironmentsService` to return real Environments
And some other easy fixes needs to be done

NikolaRusakov committed:
server get environment

karelSoupa committed:
Importing bootstrap and coreui scss from node_modules

NikolaRusakov committed:
returns  /environment
 TODO- finish EnvironmentApiController methods

NikolaRusakov committed:
Merge remote-tracking branch 'origin/Feature' into Feature

NikolaRusakov committed:
Updated environment service

TODO DBmaker for connections

NikolaRusakov committed:
Merge branch 'master' of https://github.com/petrleocompel/docker-dash into Feature

# Conflicts:
#	client/src/Cards.js
#	client/src/views/Dashboard.js

NikolaRusakov committed:
Added database connection service to EnvironmentService.java

NikolaRusakov committed:
Added environmentServiceAction method in EnvironmentService
TODO: finish the rest of actions

NikolaRusakov committed:
finished restart, stop actions of environmentServiceAction
TODO: finish the rest of actions

NikolaRusakov committed:
improved speed of environmentServiceAction
TODO: finish the rest of actions

NikolaRusakov committed:
implemented DELETE and GET methods of EnvironmentService.

NikolaRusakov committed:
DockerConnection model reworked, changes to DockerConnectionService
added get and set database connection api/controllers and methods
swagger api changed
connections.json DB changed
TODO: create selectDockerDonnection method
      consult database in depth
